src/main/java/org/homework/TodoService.java

```diff
@@ -0,0 +1,37 @@
+package org.homework;
+
+import java.time.LocalDate;
+import java.util.List;
+
+public class TodoService {
+    private TodoRepository todoRepository = new TodoRepository();
+
+    public Todo addTodo(String contents, LocalDate dueDate) {
+        return todoRepository.add(new Todo(contents, dueDate));
+    }
+
+    public Todo deleteTodo(int id) {
+        return todoRepository.delete(id)
+                .orElseThrow(() -> new IllegalArgumentException("해당 ID의 할 일이 없습니다."));
+    }
+
+    public Todo findTodoById(int id) {
+        return todoRepository.view(id)
+                .orElseThrow(() -> new IllegalArgumentException("해당 ID의 할 일이 없습니다."));
+    }
+
+    public Todo completeTodoById(int id) {
+        Todo todo = todoRepository.view(id)
+                .orElseThrow(() -> new IllegalArgumentException("해당 ID의 할 일이 없습니다."));
+        todo.setStatus(Status.완료);
```

Comment:
```suggestion
        todo.complete();
```
객체지향프로그래밍에서는 set 메소드를 사용하기 보다는 상태를 클래스 내부에서 변경하는 코드로 작성합니다.

코드가 늘어난다는 단점이 존재하지만, set메소드를 통해 객체의 상태를 외부에서 부분별하게 바뀌는것을 방지하기 위함입니다.